meson.build

```diff
@@ -0,0 +1,132 @@
+project('transmission-remote-gtk', 'c',
+        license: 'GPL2+',
+        version: '1.4.2',
+        default_options: ['c_std=gnu99',
+                          'warning_level=2',
+                          'buildtype=debugoptimized']
+)
+
+project_name = meson.project_name()
+project_version = meson.project_version()
+app_id = 'io.github.TransmissionRemoteGtk'
+datadir = get_option('datadir')
+
+# i18n module
+i18n = import('i18n')
+
+
+# glib version info
+glib_version = '2.44'
+glib_max_version = glib_version
+glib_min_version = glib_version
+
+# required dependencies
+gtk_dep     = dependency('gtk+-3.0', version: '>= 3.16')
+gio_dep     = dependency('gio-2.0', version: '>= 2.44')
+glib_dep    = dependency('glib-2.0', version: '>= @0@'.format(glib_version))
+json_dep    = dependency('json-glib-1.0', version: '>= 0.8')
+libcurl_dep = dependency('libcurl')
+gthread_dep = dependency('gthread-2.0')
+
+trg_deps = [gtk_dep, glib_dep, gio_dep, json_dep, libcurl_dep, gthread_dep]
+
+# optional dependencies
+rss_dep             = dependency('mrss', version: '>=0.18', required: get_option('mrss'))
+geoip_dep           = dependency('geoip', required: get_option('geoip'))
+libproxy_dep        = dependency('libproxy', required: get_option('libproxy'))
+libnotify_dep       = dependency('libnotify', required: get_option('libnotify'))
+libappindicator_dep = dependency('libappindicator', required: get_option('libappindicator'))
+
+opt_deps = [geoip_dep, libproxy_dep, libnotify_dep, libappindicator_dep]
+foreach opt_dep : opt_deps
+  if opt_dep.found()
+    trg_deps += opt_dep
+  endif
+endforeach
+
+# compiler
+cc = meson.get_compiler('c')
+
+# compiler deps
+trg_deps += cc.find_library('m', required: true)
```

Comment:
On systems where the math functions are part of the libc and don't exist separately, this will fail. No systems should be missing the math functions altogether -- they are either in the libc, or found as the `'m'` library.

See https://mesonbuild.com/howtox.html#add-math-library-lm-portably